src/xdp/apikernel.c

```diff
@@ -0,0 +1,539 @@
+//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+
+#include "precomp.h"
+#include "apikernel.tmh"
+
+typedef struct _XDPAPI_PROVIDER {
+    NPI_PROVIDER_CHARACTERISTICS Characteristics;
+    HANDLE NmrProviderHandle;
+} XDPAPI_PROVIDER;
+
+typedef struct _XDPAPI_CLIENT {
+    HANDLE NmrBindingHandle;
+    GUID ClientModuleId;
+    const XDP_API_CLIENT_DISPATCH *ClientDispatch;
+    const VOID *ClientBindingContext;
+} XDPAPI_CLIENT;
+
+typedef struct _XDP_API_ROUTINE {
+    _Null_terminated_ const CHAR *RoutineName;
+    VOID *Routine;
+} XDP_API_ROUTINE;
+
+static XDPAPI_PROVIDER XdpApiProvider;
+static const NPI_MODULEID NPI_XDP_MODULEID = {
+    sizeof(NPI_MODULEID),
+    MIT_GUID,
+    { 0x521421C2, 0x539B, 0x46DF, { 0xA1, 0x43, 0x8D, 0xAC, 0x88, 0xD0, 0xC1, 0x55 } }
```

Comment:
add a comment with the string version of this GUID for easier searching

Reply:
are these ascii?
`ebpf_private_extension.h` doesn't have such string.

Reply:
Put a comment for `521421C2-539B-46DF-A143-8DAC88D0C155`